homeassistant/components/twitch/sensor.py

```diff
@@ -86,71 +90,76 @@ def setup_platform(
+    async_add_entities(
+        [TwitchSensor(channel, client) for channel in twitch_users],
         True,
     )
 
 
 class TwitchSensor(SensorEntity):
-    """Representation of an Twitch channel."""
+    """Representation of a Twitch channel."""
 
     _attr_icon = ICON
 
-    def __init__(self, channel: dict[str, str], client: Twitch) -> None:
+    def __init__(self, channel: TwitchUser, client: Twitch) -> None:
         """Initialize the sensor."""
         self._client = client
         self._enable_user_auth = client.has_required_auth(AuthType.USER, OAUTH_SCOPES)
-        self._attr_name = channel["display_name"]
-        self._attr_unique_id = channel["id"]
+        self._attr_name = channel.display_name
+        self._attr_unique_id = channel.id
 
-    def update(self) -> None:
+    async def async_update(self) -> None:
         """Update device state."""
-        followers = self._client.get_users_follows(to_id=self.unique_id)["total"]
-        channel = self._client.get_users(user_ids=[self.unique_id])["data"][0]
+        if self.unique_id is None:
+            return
+        followers = (await self._client.get_users_follows(to_id=self.unique_id)).total
+        channel = await first(self._client.get_users(user_ids=[self.unique_id]))
+        if not channel:
+            return
         self._attr_extra_state_attributes = {
             ATTR_FOLLOWING: followers,
-            ATTR_VIEWS: channel["view_count"],
+            ATTR_VIEWS: channel.view_count,
         }
         if self._enable_user_auth:
-            user = self._client.get_users()["data"][0]["id"]
-
-            subs = self._client.check_user_subscription(
-                user_id=user, broadcaster_id=self.unique_id
-            )
-            if "data" in subs:
+            user = await first(self._client.get_users())
+            if not user:
+                return
+            try:
+                sub = await self._client.check_user_subscription(
+                    user_id=user.id, broadcaster_id=self.unique_id
+                )
                 self._attr_extra_state_attributes[ATTR_SUBSCRIPTION] = True
-                self._attr_extra_state_attributes[ATTR_SUBSCRIPTION_GIFTED] = subs[
-                    "data"
-                ][0]["is_gift"]
-            elif "status" in subs and subs["status"] == 404:
+                self._attr_extra_state_attributes[
+                    ATTR_SUBSCRIPTION_GIFTED
+                ] = sub.is_gift
+            except TwitchResourceNotFound:
+                _LOGGER.debug("User is not subscribed")
                 self._attr_extra_state_attributes[ATTR_SUBSCRIPTION] = False
-            elif "error" in subs:
-                _LOGGER.error(
-                    "Error response on check_user_subscription: %s", subs["error"]
-                )
-                return
-            else:
-                _LOGGER.error("Unknown error response on check_user_subscription")
+            except TwitchAPIException as exc:
```

Comment:
- Can we add a test for this case?
- Should the sensor become unavailable at this point?

Reply:
Same, I am just bumping the version at this point.